okapi-core/src/main/java/org/folio/okapi/MainVerticle.java

```diff
@@ -253,78 +250,71 @@ public void start(Promise<Void> promise) {
     final ModuleDescriptor md = InternalModule.moduleDescriptor(okapiVersion);
     final String okapiModule = md.getId();
     final String interfaceVersion = md.getProvides()[0].getVersion();
-    moduleManager.get(okapiModule).onComplete(gres -> {
-      if (gres.succeeded()) { // we already have one, go on
-        logger.debug("checkInternalModules: Already have {} "
-            + " with interface version {}", okapiModule, interfaceVersion);
-        // See Okapi-359 about version checks across the cluster
-        checkSuperTenant(okapiModule, promise);
-        return;
-      }
-      if (OkapiError.getType(gres.cause()) != ErrorType.NOT_FOUND) {
-        promise.fail(gres.cause()); // something went badly wrong
+    moduleManager.get(okapiModule).onSuccess(gres -> {
+      // we already have one, go on
+      logger.debug("checkInternalModules: Already have {} "
+          + " with interface version {}", okapiModule, interfaceVersion);
+      // See Okapi-359 about version checks across the cluster
+      checkSuperTenant(okapiModule, promise);
+    }).onFailure(cause -> {
+      if (OkapiError.getType(cause) != ErrorType.NOT_FOUND) {
+        promise.fail(cause); // something went badly wrong
         return;
       }
       logger.debug("Creating the internal Okapi module {} with interface version {}",
           okapiModule, interfaceVersion);
-      moduleManager.create(md, true, true, true).onComplete(ires -> {
-        if (ires.failed()) {
-          promise.fail(ires.cause()); // something went badly wrong
-          return;
-        }
+      moduleManager.create(md, true, true, true).onFailure(cause1 ->
+          promise.fail(cause1) // something went badly wrong
+      ).onComplete(ires -> {
```

Comment:
Is this supposed to be `onSuccess` or `onComplete`? Do you want `checkSuperTenant` always be done?

Reply:
It should be onSuccess  (and that was the existing behavior).